saleor/graphql/order/types.py

```diff
@@ -388,6 +396,31 @@ def resolve_transaction(
             granted_refund.transaction_item_id
         )
 
+    @staticmethod
+    def resolve_reason_reference(
+        root: SyncWebhookControlContext[models.OrderGrantedRefund], info
+    ):
+        if not root.node.reason_reference:
+            return None
+
+        def wrap_page_with_context(page):
+            if not page:
+                return None
+
+            return (
+                ChannelByOrderIdLoader(info.context)
+                .load(root.node.order_id)
+                .then(
+                    lambda channel: ChannelContext(node=page, channel_slug=channel.slug)
+                )
+            )
+
+        return (
+            PageByIdLoader(info.context)
+            .load(root.node.reason_reference_id)
+            .then(wrap_page_with_context)
+        )
```

Comment:
Non blocking:
You can fetch Page and Channel together:
```suggestion
    def resolve_reason_reference(
        root: SyncWebhookControlContext[models.OrderGrantedRefund], info
    ):
        if not root.node.reason_reference:
            return None

        def with_page_and_channel(page, channel):
            if not page:
                return None

            return (
            ChannelContext(node=page, channel_slug=channel.slug)
                )
            )

        page = PageByIdLoader(info.context).load(root.node.reason_reference_id)
        channel = ChannelByOrderIdLoader(info.context).load(root.node.order_id)
        return Promise.all(
            [page, channel]
        ).then(with_page_and_channel)
```